src/Extensions/Hosting/ExceptionLogging/ExceptionLoggingHostedService.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.Hosting
+{
+	internal class ExceptionLoggingHostedService : IHostedService
+	{
+		private readonly ILogger<ExceptionLoggingHostedService> m_logger;
+
+		public ExceptionLoggingHostedService(ILogger<ExceptionLoggingHostedService> logger)
+			=> m_logger = logger;
+
+		public Task StartAsync(CancellationToken cancellationToken)
+		{
+			AppDomain.CurrentDomain.FirstChanceException += FirstChanceException;
+			return Task.CompletedTask;
+		}
+
+		public Task StopAsync(CancellationToken cancellationToken)
+		{
+			AppDomain.CurrentDomain.FirstChanceException -= FirstChanceException;
+			return Task.CompletedTask;
+		}
+
+		private void FirstChanceException(object sender, FirstChanceExceptionEventArgs e)
+		{
+			m_logger.LogError(Tag.Create(), e.Exception, "Exception during application execution");
```

Comment:
Have you actually tested it? I have a feeling it's not going to end well. You'll be logging EVERY exception, including the expected and handled ones within .NET.

Reply:
.net code if quite good in not throwing and catching exceptions instead of logic, but after testing it looks like SF still produces a lot of COM exceptions, so you are probably right. Although I still think it would be useful to know about those, for performance reasons, we could probably remove this for now.
I've changed the approach to use host wrapper, what do you think about this option?